### PR TITLE
Better define a redeploy

### PR DESCRIPTION
**Pull request by chadwcarlson**

### CODE REVIEW

File: docs/src/bestpractices/environment-build.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ For variables that should vary between production and "other" environments, such
 
 ### Static file configuration
 
-A few applications, unfortunately, require configuration values to be specified in a static, non-executable file (such as a `.ini`, `.xml`, or `.yaml` file) and do not support reading from environment variables.  These files cannot be populated at build time as environment-specific values are not available, but cannot be written to in deploy as the file system is read only.  That is a design flaw in the application, and you should file a bug with the application or framework author.
+A few applications, unfortunately, require configuration values to be specified in a static, non-executable file (such as a `.ini`, `.xml`, or `.yaml` file) and do not support reading from environment variables.  These files cannot be populated at build time as environment-specific values are not available, but cannot be written to in deploy as the file system is read only.
```
Comment (MariadeAnton) [inline docs/src/bestpractices/environment-build.md:76]:
```suggestion
A few applications, require configuration values to be specified in a static, non-executable file (such as a `.ini`, `.xml`, or `.yaml` file) and do not support reading from environment variables.  These files cannot be populated at build time as environment-specific values are not available, but cannot be written to in deploy as the file system is read only.
```

File: docs/src/configuration/app/build.md [hunk 2]
```diff
@@ -116,6 +116,10 @@ Performed update: my_custom_profile_update_7001
 Finished performing updates.
 ```
 
+Your `deploy` hook is tied to commits in the same way as your builds. That is, once a commit has been pushed and a new build image has been created, both the `build` and `deplo`y hooks will be resused during operations that are not caused by subsequent commits.
```
Comment (MariadeAnton) [inline docs/src/configuration/app/build.md:119]:
```suggestion
Your `deploy` hook is tied to commits in the same way as your builds. That is, once a commit has been pushed and a new build image has been created, the result of both `build` and `deploy` hooks will be resused, unless there's a new git commit. 
```

File: docs/src/configuration/app/build.md [hunk 3]
```diff
@@ -116,6 +116,10 @@ Performed update: my_custom_profile_update_7001
 Finished performing updates.
 ```
 
+Your `deploy` hook is tied to commits in the same way as your builds. That is, once a commit has been pushed and a new build image has been created, both the `build` and `deplo`y hooks will be resused during operations that are not caused by subsequent commits. 
+
+This means that adding variables, changing access permissions, or even running a `redeploy` using the CLI or management console will not cause the `deploy `hook to run again for the current commit. Redeploys trigger only the `post_deploy` hook to run again from the beginning, and a committed change to the application is needed to rerun the `build` and `deploy` hooks.
```
Comment (MariadeAnton) [inline docs/src/configuration/app/build.md:121]:
```suggestion
Redeploys trigger only the `post_deploy` hook to run again from the beginning, and a committed change to the application is needed to rerun the `build` and `deploy` hooks.
This means that adding variables, changing access permissions, or triggering a `redeploy` using the CLI or management console will not cause the `deploy `hook to run again for the current commit. 
```

File: docs/src/development/troubleshoot.md [hunk 4]
```diff
@@ -19,6 +19,12 @@ platform redeploy
 
 Please note that the redeploy will happen after any scheduled builds in either "Running" or "Pending" state. 
 
+{{< note >}}
+Despite the name, triggering a redeploy will not cause the `deploy` hook to rerun for your application. Both your builds and `deploy` hook are tied to individual commits and are resused until another commit is pushed to the environment.
```
Comment (MariadeAnton) [inline docs/src/development/troubleshoot.md:23]:
```suggestion
Despite the name, triggering a redeploy will not cause the `deploy` hook to rerun for your application. Both your `build` and `deploy` hook are tied to individual commits and are resused until another commit is pushed to the environment. 
```

**Review by MariadeAnton — APPROVED** [review]

File: docs/src/configuration/app/build.md [hunk 5]
```diff
@@ -124,6 +130,8 @@ What is "safe" to run in a `post_deploy` hook vs. in a `deploy` hook will vary b
 
 The `post_deploy` hook logs to its own file in addition to the activity log, `/var/log/post-deploy.log`.
 
+The `post_deploy` hook is the only hook provided that will run from the beginning during a redeploy.
```
Comment (pjcdawkins) [inline docs/src/configuration/app/build.md:133]: @chadwcarlson @MariadeAnton 

> from the beginning

I guess you're saying it runs "every time"? I'm not sure what the beginning refers to

Comment (pjcdawkins) [reply docs/src/configuration/app/build.md:133]: ... googled.. maybe that's a false friend of _d'emblée_? :fr:

**Pull request merged**